### PR TITLE
Pll sys freq bug

### PR DESCRIPTION
**Pull request by m-bossart**
The frequency from the PLL (both `KauraPLL` and `ReducedOrderPLL`) was modeled with respect to 1.0, regardless of the frequency reference used in the simulation.  Also includes a modification of the documentations for VSM to be more consistent with the documentation of the Droop outer control.

### CODE REVIEW

**Comment by jd-lara** [comment]
@m-bossart we have tested this and at times it can be a tricky thing. I am ok with it but a review from @rodrigomha is also important.

File: src/models/inverter_models/frequency_estimator_models.jl [hunk 1]
```diff
@@ -62,7 +62,7 @@ function mdl_freq_estimator_ode!(
 
     #Update inner_vars
     #PLL frequency, D'Arco EPSR122 eqn. 16
-    inner_vars[ω_freq_estimator_var] = Δω_pi + 1.0
+    inner_vars[ω_freq_estimator_var] = Δω_pi + ω_sys
```
Comment (jd-lara) [inline src/models/inverter_models/frequency_estimator_models.jl:65]: @m-bossart we need to test this properly. A simulation with the trace will help.

Comment (m-bossart) [reply src/models/inverter_models/frequency_estimator_models.jl:65]: Are you referring to a PSCAD simulation for comparison?

Comment (m-bossart) [reply src/models/inverter_models/frequency_estimator_models.jl:65]: Or just a PSID simulation where` w_sys != 1.0`  ?

Comment (jd-lara) [reply src/models/inverter_models/frequency_estimator_models.jl:65]: Just a PSID simulation pre and post change. Just drop the plots here please

**Comment by rodrigomha** [comment]
I think it is okay being consistent with using w_sys. My main concern is that this is not well documented anywhere. When they add 1.0 in most of the control blocks, I'm not sure if they are adding a reference frequency, or is it 1.0 because they assume that the network SRF is rotating at constant frequency = 1.0.

**Comment by m-bossart** [comment]
Is there a specific reference you used when creating this model? In the PSID docs there is a sentence: "The notation used a δθ refers as the variation of the respective angle θgrid with respect to the grid SRF".  That make me think that ` ω = Δω_pi + ω_sys`

**Comment by rodrigomha** [comment]
> Is there a specific reference you used when creating this model? In the PSID docs there is a sentence: "The notation used a δθ refers as the variation of the respective angle θgrid with respect to the grid SRF". That make me think that ` ω = Δω_pi + ω_sys`

That sentence is from D'Arco paper, so what you are saying is what most likely make sense.

File: src/models/inverter_models/frequency_estimator_models.jl (hunk 1 above)
Comment (m-bossart) [reply src/models/inverter_models/frequency_estimator_models.jl:65]: Ok I ran simulations pre and post change on the three bus system with two inverters -- VSM at bus 101 (the reference bus) and Droop at bus 102. Here are the results before the change. I've included the results from a PSCAD simulation as well. The key takeaway is that the steady state result of each simulation (PSCAD vs PSID) is completely different due to the issue with the PLL relative reference frame. 
Voltage
Frequency
P
Q

The results after the proposed change is below: Now the steady state behavior matches --- the settling values in terms of P,Q, V, freq are the same for PSID and PSCAD. Obviously there are some large oscillations in the PSID results that are not present in PSCAD. I believe these are fundamentally related to the modelling using a synchronous reference frame that is required to have a time-invariant simulation in PSID. This is an issue to dig into further but I believe the proposed change is correct even though it results in this oscillatory behavior for this system. 
Voltage_bugfix
Frequency_bugfix
P_bugfix
Q_bugfix

Comment (m-bossart) [reply src/models/inverter_models/frequency_estimator_models.jl:65]: The fact that the choice of SRF can lead to oscillations is consistent with the fact that the system is small signal unstable if we choose w_sys :=1.0. I think the impact of modelling the network in a single reference frame is underexplored in the literature.

Comment (jd-lara) [reply src/models/inverter_models/frequency_estimator_models.jl:65]: Ok, yeah this is a good change. Thanks for taking the challenge of validating, this is a huge contribution. 

About the oscillations, I don't agree really with "I believe these are fundamentally related to the modelling using a synchronous reference frame that is required to have a time-invariant simulation in PSID. This is an issue to dig into further but I believe the proposed change is correct even though it results in this oscillatory behavior for this system." 

I think that this is about the difference in how the input frequency is measured  in both models and how that inputs affects the device models.

Comment (m-bossart) [reply src/models/inverter_models/frequency_estimator_models.jl:65]: Results with both IDA and Rodas5 below. The traces are overlaid. All of these results are for low tolerance (1e-14) and small timestep (5 microseconds) in PSID/PSCAD respectively to try to eliminate numerical differences and focus on modeling issues. 
Frequency_IDA
P_IDA
Q_IDA
Voltage_IDA

Comment (jd-lara) [reply src/models/inverter_models/frequency_estimator_models.jl:65]: ok, thanks for doing this. I feel comfortable with these results. Lets chat later today about the SRF and merge once we iron out those details.

**Comment by m-bossart** [comment]
Angle of the VSM pll before and after this change: 
plot_4

**Comment by jd-lara** [comment]
Thanks a lot to @m-bossart for this important fix. I will merge as soon as the tests pass.

**Pull request merged**